ethereum/contracts/ShutdownSwitch.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+import "./interfaces/IWormhole.sol";
+
+contract ShutdownStructs {
+    struct VoterEntry {
+        address key;
+        bool enabled;
+    }
+}
+
+abstract contract ShutdownSwitch {
+    function requiredVotesToDisable() public view returns (uint16) {
+        return requiredNoVotes;
+    }
+
+    function numVotesToDisable() public view returns (uint16) {
+        return votesToDisable;
+    }
+
+    function enabledFlag() public view returns (bool){
+        return enabled;
+    }
+
+    event ShutdownVoteCast(address indexed voter, bool votedToEnable, uint16 numVotesToDisable, bool enableFlag);
+    event ShutdownStatusChanged(bool enabledFlag, uint16 numVotesToDisable);
+
+    function getWH() public virtual view returns (IWormhole);
+    function getChainId() public virtual view returns (uint16);
+
+    uint16 constant REQUIRED_NO_VOTES = 1;
+
+    ShutdownStructs.VoterEntry[20] voters;
```

Comment:
Reading through the code, there's a lot of looping around this array, it looks a little error-prone. How about we represent this instead as
```solidity
mapping(address => bool) voters;
```
then for counting up the votes, we just iterate through the keys in the current guardian set and see what the guardian is set to.